### PR TITLE
separate PluginId module from config.rb

### PR DESCRIPTION
**Pull request by tagomoris**
* simply speaking, it's wrong file to write it
* added a feature to check whether id is intentionally set by user or not
  * it's important to check whether we can create many thing depends on id or not
  * for example, id-based auto generated buffer path, and plugin storage on local filesystem

### CODE REVIEW

File: lib/fluent/plugin_id.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+module Fluent
+  module PluginId
+    @@configured_ids = {}
```
Comment (repeatedly) [inline lib/fluent/plugin_id.rb]: It seems stored values are not used.
Using Set or Array is better for memory usage and GC pressure.

Comment (tagomoris) [inline lib/fluent/plugin_id.rb]: Ah, that's right. I'll use it.

**Comment by tagomoris** [comment]
I modified commits:
* to use Set instead of Hash
* to stringify `@id` not to duplicate values between strings and numbers (or other types)
* not to add nil if `@id` is missing

**Comment by tagomoris** [comment]
Merged.

**Comment by sonots** [comment]
LGTM

**Pull request merged**